src/libostree/ostree-repo-commit.c

```diff
@@ -493,6 +493,33 @@ _ostree_repo_bare_content_write (OstreeRepo                 *repo,
 {
   OstreeRealRepoBareContent *real = (OstreeRealRepoBareContent*) barewrite;
   g_assert (real->initialized);
+
+  if ((repo->min_free_space_percent > 0 || repo->min_free_space_size > 0) && repo->in_transaction)
+    {
+      g_mutex_lock (&repo->txn_lock);
+      g_assert_cmpint (repo->txn.blocksize, >, 0);
+
+      struct stat st_buf;
+      if (!glnx_fstat (real->tmpf.fd, &st_buf, error))
```

Comment:
During testing out this patch today, I realized the `st_size` of real->tmpf.fd , is turning out to be 0. Looks like it has to do something with O_TMPFILE. Investigating.

Reply:
>Thinking more on this, I'm not sure if this is the best place to check for this since we're always rounding up on each write so we might be greatly overestimating the space a new object takes once we're done with it.

Can you explain this a bit more. I am not sure if I follow. In my observation, when there is a static delta fetch (1-part), there are multiple calls to `_ostree_repo_bare_content_commit` . I thought it would be just at-most one call to `_ostree_repo_bare_content_commit` to commit. Maybe my knowledge of deployment from static-deltas is lacking.

Reply:
Each delta part can have instructions for building multiple files. Check out `_ostree_static_delta_part_execute`. For each content object that gets created, there is a call to `_ostree_repo_bare_content_open`, one or more calls to `_ostree_repo_bare_content_write`, and then a final call to `_ostree_repo_bare_content_commit`. I'm simplifying a bit though (there's different paths for e.g. metadata objects and for committing to archive repos).